Dockerfile

```diff
@@ -85,6 +85,7 @@ ENV LANG=C.UTF-8 \
     SUPERSET_PORT=8088
 
 RUN useradd --user-group -d ${SUPERSET_HOME} --no-log-init --shell /bin/bash superset \
+        && mkdir -p ${SUPERSET_HOME} \
```

Comment:
Why not use `useradd -m`. 
```
-m, --create-home             create the user's home directory
```

Reply:
because /app only allows `root` and superset user is not created yet, so that fails

Reply:
if fails with:
```
 => ERROR [lean  2/11] RUN useradd --user-group -m -d /app/superset_home --no-log-init --shell /bin/bas  0.3s
------
 > [lean  2/11] RUN useradd --user-group -m -d /app/superset_home --no-log-init --shell /bin/bash superset         && mkdir -p /app/pythonpath         && apt-get update -y         && apt-get install -y --no-install-recommends             build-essential             default-libmysqlclient-dev             libsasl2-modules-gssapi-mit             libpq-dev         && rm -rf /var/lib/apt/lists/*:
#6 0.304 useradd: cannot create directory /app/superset_home
```

Reply:
:point_up: is caused by missing directory `/app`, which comes after this line but really should put before it:

https://github.com/apache/superset/blob/0eb34be78937edc509b6df872bc831d28add675f/Dockerfile#L88

see my PR: #14903

Reply:
Got it.
So I'd like to suggest that you need to change ownership of `${SUPERSET_HOME}` to `superset` as well.
```
chown -R superset: ${SUPERSET_HOME}`
```

Reply:
@shawnzhu's https://github.com/apache/superset/pull/14903 is in line with my testing for https://github.com/apache/superset/issues/14847#issuecomment-848961017 and avoids the issue with home directory ownership present in this PR.

Reply:
@shawnzhu @ethack agree it's the missing PYTHONPATH. let's merge #14903, thank you